project-management/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java

```diff
@@ -97,10 +97,15 @@ private Project createProject(String code, String title, String objective,
 
     var registrationResult = projectDomainService.registerProject(intent, projectCode, projectManager, principalInvestigator, responsiblePerson, funding);
 
-    if (registrationResult.isError() && registrationResult.getError().equals(ResponseCode.PROJECT_REGISTRATION_FAILED)) {
-      throw new ApplicationException("Project registration failed.", ErrorCode.GENERAL, ErrorParameters.of(code));
-    }
-
+    registrationResult.onError(responseCode -> {
+      if (responseCode.equals(ResponseCode.PROJECT_ALREADY_EXISTS)) {
```

Comment:
In Java you can now use swich expressions to ensure you handle every response code here.

Reply:
https://docs.oracle.com/en/java/javase/17/language/switch-expressions.html#GUID-BA4F63E3-4823-43C6-A5F3-BAA4A2EF3ADC

Reply:
Sadly the data-manager-application is running in java 17 and pattern matching in switch statements is only possible in java 21 :( 
Additionally since the domain service only provides 2 error responses (Registration failed or Already Exists) an if else statement is not more code 🤔

Reply:
No for enums this exists from java 17 onwards :)